Frontend/ui-driver/src/Screens/DriverProfileScreen/Controller.purs

```diff
@@ -257,12 +262,33 @@ data Action = BackPressed
             | ShareQR PrimaryButtonController.Action
             | ManageVehicleButtonAC PrimaryButtonController.Action
             | PendingVehicle String ST.VehicleCategory
+            | CompleteProfile 
             | OpenCancellationRateScreen
+            | ProfileDataAPIResponseAction DriverProfileDataRes
 
 eval :: Action -> DriverProfileScreenState -> Eval Action ScreenOutput DriverProfileScreenState
 
+eval (ProfileDataAPIResponseAction res) state = do 
+  let DriverProfileDataRes resp = res 
+      _ = spy "res is" res
+      pledgeOthers = (DA.filter (\item -> (item /= "Safe Journey" && item /= "Clean Car" && item /= "On-time Pick up" && item /= "Regular Maintenance") )) resp.pledges
+      whyNyOthers = spy "whyNyOthers" $ (DA.filter (\item -> (item /= "Buy New Home" && item /= "Kid's Education" && item /= "Buy New Vehicle") )) resp.aspirations
+  continue state{data{res{
+    completed = getValueBtwRange ((if resp.pledges == [] then 0 else 1) + (if resp.aspirations == [] then 0 else 1) + (if resp.drivingSince == Nothing then 0 else 1) + (if resp.hometown == Nothing then 0 else 1) + (if resp.vehicleTags == [] then 0 else 1) + (if resp.otherImageIds == [] then 0 else 1)) 0 6 0 4,
+    pledge = (DA.filter (\item -> (item == "Safe Journey" || item == "Clean Car" || item == "On-time Pick up" || item == "Regular Maintenance") )) resp.pledges, 
```

Comment:
No string comparison

Reply:
same thing we are not using this these strings only for comparison but to update the state also